scripts/genCurrencies.mjs

```diff
@@ -0,0 +1,76 @@
+/* eslint-disable functional/no-expression-statement */
+
+/**
+ * @typedef {import('./getCurrencyData.mjs').CurrencyData} CurrencyData
+ *
+ * @callback formatFn
+ * @param {CurrencyData} currency - The currency data to transform into a source file.
+ * @returns {string} - The generated currency source file.
+ */
+
+import { promises as fs } from 'fs';
+import path from 'path';
+
+// eslint-disable-next-line import/default
+import prettier from 'prettier';
+
+import { getCurrencyData } from './getCurrencyData.mjs';
+
+// prettier requires this to import into an es module without error.
+// eslint-disable-next-line import/no-named-as-default-member
+const { format, resolveConfig } = prettier;
+
+const dirExists = (dirPath) =>
+  fs
+    .stat(dirPath)
+    .then((stats) => stats.isDirectory())
+    .catch(() => false);
+
+/**
+ * Generate a set of currency source files.
+ *
+ * @param {string} outputDir - The directory to output the source files into.
+ * @param {formatFn} formatFn - The format function used to generate individual currency files.
+ * @returns {Promise<void>} - Returns a Promise that resolves when all the files are generated.
+ */
+export async function genCurrencies(outputDir, formatFn) {
```

Comment:
Because it confused me:
`format` might not be the ideal name here.
I got confused, as  to why there are to functions named `format` in this function being used.
One being prettier, the other one being a generator of code.

I would rename this to `generatefFn` instead, to be extra clear

Reply:
`generateFn` sounds like a much better name, I will switch to that.